### PR TITLE
Add: Monthly Dev Post of April 2019

### PR DESCRIPTION
**Pull request by TrueBrain**
[Rendered](https://github.com/OpenTTD/website/blob/monthly-dev-post/_posts/2019-04-01-monthly-dev-post.md)

### CODE REVIEW

**Comment by nielsmh** [comment]
I don't have write access to this repository, so here's some maybe-useful notes:

- Comment where @ldpl attached the 50k stations save: https://github.com/OpenTTD/OpenTTD/pull/7235#issuecomment-465280438
- There is also the Wentbourne save. I'm quite sure it's sourced from the Savegames forum on TT-Forums, but couldn't find the original thread last I looked.

- The two most interesting performance gains from the k-d tree patch directly are probably map panning performance with a large number of viewport signs, and AI speed gains from tile-town lookups. It also opens up the possibility of more new features depending on tile-town lookups being fast.

- Write something about how old "exploits" are fixed, and generate some discussion about whether they were really exploits. (Sparse station catchment areas, and building rail out to oil rigs.)

**Comment by TrueBrain** [comment]
You should have write access to this repository (in fact, to all OpenTTD repositories). It even says so next to your name ;) So I suggest we try to fix that ;)

Are you sure you don't have write access? Where did it fail?

https://github.com/OpenTTD/website/blob/monthly-dev-post/_posts/2019-04-01-monthly-dev-post.md
There is an Edit icon on the top right. You can click that, make the changes, and commit. Or at least, you should be able. Let me know where you get stuck, and let's take it from there!

File: _posts/2019-04-01-monthly-dev-post.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+DJGPP (the compiler we use to create DOS binaries) doesn't have support for that.
+Last month we created some test-binaries to see the current state of DOS.
+Turns out, the performance is not good (<1fps).
+Considering all these things together, we strongly wonder if having DOS support is worth it.
+So we ask you: would anyone mind of we drop DOS support, so we can use modern techniques to build OpenTTD?
+Let us know on the forum thread what you think!
+
+### MorphOS / AmigoOS / BeOS support
+
+In 1.10 we will be dropping support for MorphOS and AmigoOS, as well as BeOS (you can use Haiku instead of BeOS).
```
Comment (nikolas) [inline _posts/2019-04-01-monthly-dev-post.md]: AmigoOS -> AmigaOS

Comment (TrueBrain) [reply _posts/2019-04-01-monthly-dev-post.md]: Gave you write permission, so you can fix that yourself :D At least I was consistent in doing it wrong :P

Comment (nikolas) [reply _posts/2019-04-01-monthly-dev-post.md]: Hmmm.. the Edit button is still grey. 
2019-03-11-100133_600x64_scrot

Comment (TrueBrain) [reply _posts/2019-04-01-monthly-dev-post.md]: Ah, yes, you have an invite in your mailbox you first have to accept; seems I cannot force you to join our group :D

Comment (TrueBrain) [reply _posts/2019-04-01-monthly-dev-post.md]: Ty!!

**Comment by andythenorth** [comment]
Are we talking about nmlc as part of OpenTTD stuff? 

16-cargo support has come up in forums a bit: https://github.com/OpenTTD/nml/pull/15

**Comment by PeterN** [comment]
@andythenorth I don't see why not. It's now being maintained under our umbrella.

File: _posts/2019-04-01-monthly-dev-post.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+If you feel up to it, and can maintain those targets for a longer period, we would love to see a Pull Request reintroducing them.
+You can simply revert the patch that removed the code, and work from there.
+
+## NewGRF
+
+### Tools
+
+NML gained support for 16 cargoes which was added to OpenTTD earlier.
+For nml projects it means that they need to update the code for the production callback.
+Check FIRS code for an example.
```
Comment (Eddi-z) [inline _posts/2019-04-01-monthly-dev-post.md]: I don't think that is a good sentence to put in a dev post like this. be more elaborate?

Comment (planetmaker) [reply _posts/2019-04-01-monthly-dev-post.md]: I totally agree that it should be more elaborate. Yet better a short version than no version.

File: _posts/2019-04-01-monthly-dev-post.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+It had some serious drawbacks: no threads, no network, different video-driver (allegro instead of SDL).
+This created some huge technical debt.
+We have tons of places where we have to keep in mind that network might also not exist (and DOS is the only one without network support).
+We have special glue to handle threads; and the compiler doesn't follow std::thread (C++11), so we can't switch to that.
+Additionally, we are working towards CMake support.
+DJGPP (the compiler we use to create DOS binaries) doesn't have support for that.
+Last month we created some test-binaries to see the current state of DOS.
+Turns out, the performance is not good (<1fps).
+Considering all these things together, we strongly wonder if having DOS support is worth it.
+So we ask you: would anyone mind of we drop DOS support, so we can use modern techniques to build OpenTTD?
```
Comment (glx22) [inline _posts/2019-04-01-monthly-dev-post.md]: It's already drop, would be strange to ask now

**Pull request merged**